### PR TITLE
NextJs docs has been added 

### PR DESCRIPTION
**Pull request by Aksshay88**
## Related Issue

#359 

## Description

I have added the Next js docs from the scratch where user could see from the first how to download the Nextjs and app routing ,page routing , architecture 

## Type of PR

- [X] Feature enhancement

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]
image
image

## Checklist

- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)
- [NextJs Documetation](https://nextjs.org/docs) 

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by Aksshay88** [comment]
Pls do add the GSSOC Label

File: docs/NextJs/_category_.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "label": "Next",
+    "position": 8,
```
Comment (Ajay-Dhangar) [inline docs/NextJs/_category_.json:3]: `"position": 8,`   already define you can try to put other `position`

File: docs/NextJs/index.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<div align=center>
```
Comment (Ajay-Dhangar) [inline docs/NextJs/index.md:1]: use on top

```md
---
id: ____
title: ____
sidebar_label: ____
sidebar_position: ____
tags: [ ____, ____ ]
description: ____
---
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Make Sure, You follow our structure

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)
- [NextJs Documetation](https://nextjs.org/docs) 

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

**Comment by Aksshay88** [comment]
@Ajay-Dhangar Sorry For making it a mistake Now I resolved the changes you have mentioned .You could check it now .Thanks in advance.

File: docs/NextJs/index.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+---
+id: Next JS
```
Comment (Ajay-Dhangar) [inline docs/NextJs/index.md:2]: id put like `next-js`

Comment (Ajay-Dhangar) [inline docs/NextJs/index.md:2]: id put like `next-js`

File: docs/NextJs/index.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+}
+```
+Now, you can start the development server:
+
+Run `npm run dev` to start the development server.
+Visit `http://localhost:3000` to view your application.
+Edit `app/page.tsx` (or `pages/index.tsx`) file and save it to see the updated result in your browser.
+
+
+<a href="https://nextjs.org/docs/app"><img src="https://github.com/Aksshay88/Macbook/assets/119944779/1c389169-463a-42e1-b4f0-28830ed4b8a8"></a>
```
Comment (Ajay-Dhangar) [inline docs/NextJs/index.md]: **Bad Practice:**

```md
<img src="img.png">
```

**Good Practice:**

```md
<img src="img.png" alt="Image name" />
```

**Review by Aksshay88 — COMMENTED** [review]
I have changed the suggestion you have asked for .Thanks in advance.Happy Hacking

**Comment by Aksshay88** [comment]
@Ajay-Dhangar  Just now saw what the problem is that docker cli is not identifying my image from the format of webp

Now I have added it in a local format where it could identify the image and it could run without error 

I request you to run it again or review the changes I have made.Thank you

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
change for best practice

**Comment by Aksshay88** [comment]
@Ajay-Dhangar sure will change it and get back to ASAP

Thanks in advance

File: docs/NextJs/index.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+Would you like to use Tailwind CSS? No / Yes
+Would you like to use `src/` directory? No / Yes
+Would you like to use App Router? (recommended) No / Yes
+Would you like to customize the default import alias (@/*)? No / Yes
+What import alias would you like configured? @/*
+
+```
+After the prompts, `create-next-app` will create a folder with your project name and install the required dependencies.
+
+Create a root layout inside `app/layout.tsx` with the required `<html>` and ``<body>`` tags:
```
Comment (Ajay-Dhangar) [inline docs/NextJs/index.md:34]: **Bade Practice:**

```md
``<body>``
```

**Good Practice:**

```md
`<body>`
```

File: docs/NextJs/index.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+}
+```
+Now, you can start the development server:
+
+Run `npm run dev` to start the development server.
+Visit `http://localhost:3000` to view your application.
+Edit `app/page.tsx` (or `pages/index.tsx`) file and save it to see the updated result in your browser.
+
+
+<a href="https://nextjs.org/docs/app"><img src="emty.png"></a>
```
Comment (Ajay-Dhangar) [inline docs/NextJs/index.md:66]: **Bad Practice:**

```md
<img src="empty.png">
```

**Good Practice:**

```md
<img src="empty.png" />
```

In JSX, all tags must be self-closing, so it's important to add the closing slash to the `` tag.

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Everything is fine. Please focus on XML/JSX syntax best practices.

**Comment by Aksshay88** [comment]
@Ajay-Dhangar Completed changing it .I hope your suggestion that you have requested has been rectified and thanks in advance .

**Review by Ajay-Dhangar — APPROVED** [review]
Done and fixed issues but content is not good so, next time write more better.

**Comment by Aksshay88** [comment]
@Ajay-Dhangar Sure Next time will write the content in a detailed manner

**Pull request merged**